scripts/base/frameworks/telemetry/main.zeek

```diff
@@ -0,0 +1,505 @@
+
+	## Increment a Counter through the CounterFamily
+	## Using a negative amount is an error.
+	global counter_family_inc: function(c: CounterFamily,
+	                                    label_values: labels_vector &default=vector(),
+	                                    amount: double &default=1.0): bool;
+
+	## Increment a Counter through the CounterFamily
+	## Using a negative amount is an error.
+	global counter_family_inc_t: function(c: CounterFamily,
+	                                      labels: labels_table &default=table(),
+	                                      amount: double &default=1.0): bool;
+
+	## Helper to set a Counter through the CounterFamily.
+	global counter_family_set: function(c: CounterFamily,
+	                                    value: double,
+	                                    label_values: labels_vector &default=vector()): bool;
+
+	## Helper to set a Counter through the CounterFamily.
+	global counter_family_set_t: function(c: CounterFamily,
+	                                      value: double,
+	                                      labels: labels_table &default=table()): bool;
+
+	type GaugeFamily: record {
+		__family: opaque of dbl_gauge_metric_family;
+		__labels: vector of string;
+	};
+
+	type Gauge: record {
+		__metric: opaque of dbl_gauge_metric;
+	};
+
+	## Register a gauge family.
+	global register_gauge_family: function(opts: MetricOpts): GaugeFamily;
+
+
+	## Get a handle to a Gauge of the given family and label values.
+	global gauge_with: function(gf: GaugeFamily,
+	                            label_values: labels_vector &default=vector()): Gauge;
+
+	## Get a Gauge instance from a GaugeFamily by providing
+	## label values as a table.
+	global gauge_with_t: function(gf: GaugeFamily,
+	                              labels: labels_table &default=table()): Gauge;
+
+	## Increment a gauge by the given amount.
+	global gauge_inc: function(g: Gauge, amount: double &default=1.0): bool;
+	## Decrement a gauge by the given amount.
+	global gauge_dec: function(g: Gauge, amount: double &default=1.0): bool;
+	## Set a gauge to the given value.
+	global gauge_set: function(g: Gauge, value: double): bool;
+
+	## Increment a Gauge by the given amount through a GaugeFamily.
+	global gauge_family_inc: function(g: GaugeFamily,
+	                                  label_values: labels_vector &default=vector(),
+	                                  amount: double &default=1.0): bool;
+	## Decrement a Gauge by the given amount through a GaugeFamily.
+	global gauge_family_dec: function(g: GaugeFamily,
+	                                  label_values: labels_vector &default=vector(),
+	                                  amount: double &default=1.0): bool;
+	## Set a Gauge by the given amount through a GaugeFamily.
+	global gauge_family_set: function(g: GaugeFamily,
+	                                  value: double,
+	                                  label_values: labels_vector &default=vector()): bool;
```

Comment:
Some API functions receive the labels before the value, some after. Would be good to make the order consistent.

Reply:
These were purposefully "inconsistent" so that `&default`'s work out.

The `_inc` versions have `label_values` as the second argument and `amount` as third because it's more likely that `label_values` is to be varied/set. In the `gauge_family_set()` case, the `value` field is required as there's no reasonable default, while the `label_values` may be left default (empty), that's why I moved the required one forward.

That said, consistent order is probably better. IMO `_inc()` should continue to be what it is and default to `amount=1.0` with labels being more likely to vary. For the `_set()` case that means `label_values` will become a required second parameter.

I'll make the change.